dashboard/app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -190,10 +205,9 @@ def sign_in_user
     sign_in_and_redirect @user
   end
 
-  def allows_silent_takeover(oauth_user)
+  def allows_silent_takeover(oauth_user, auth_hash)
     allow_takeover = oauth_user.provider.present?
-    allow_takeover &= %w(facebook google_oauth2 windowslive).include?(oauth_user.provider)
-    # allow_takeover &= oauth_user.email_verified # TODO (eric) - set up and test for different providers
+    allow_takeover &= %w(facebook google_oauth2 windowslive).include?(auth_hash.provider.to_s)
```

Comment:
should these be constants in `AuthenticationOption`? we have `OAUTH_CREDENTIAL_TYPES` and `CREDENTIAL_TYPES` there, so it might be nice to have `TAKEOVER_CREDENTIAL_TYPES` (or something along those lines)